api/src/main/java/ca/bc/gov/educ/api/grad/report/service/CommonService.java

```diff
@@ -569,4 +573,14 @@ public ResponseEntity<InputStreamResource> getStudentCredentialByType(UUID stude
 		}
 		return null;
 	}
+
+	private boolean isClobDataChanged(String currentBase64, String newBase64) {
+		if (currentBase64 == null || newBase64 == null) {
+			return true;
+		}
+		if (currentBase64.length() != newBase64.length()) {
+			return true;
+		}
+		return !currentBase64.equals(newBase64);
+	}
```

Comment:
Hi @infstar, I think rather than using length to compare the two values, we might consider using a hash function to create a unique signature of each value and compare those. There are some great libraries available. More info here: https://www.baeldung.com/sha-256-hashing-java